### PR TITLE
feat(logging): convergence logging now in intentActivityRepo

### PR DESCRIPTION
**Pull request by emjburns**
In order to make logging more manageable for each intent for each convergence, I've moved everything but the `com.netflix.spinnaker.q.QueueProcessor   : [] Received message ConvergeIntent(` messages to the intentActivityRepository. 

Hitting `v1/intents/{intentId}/convergenceLog` will give you all the messages for each intent with the format 
```{
"intentId": "Application:emilykeeltest",
"changeType": "NO_CHANGE",
"orchestrations": [],
"messages": [
  "System state matches desired state"
],
"diff": [],
"actor": "keel:scheduledConvergence",
"timeMs": 1516214128706
}
```

Each message is identified by its timestamp, which is created when the log message is created in `OrcaIntentLauncher`. This message is created right after orchestrations are launched in orca.

`v1/intents/{intentId}/convergenceLog/{timeMs}` is a permalink to the specific log message, while it exists. 

The number of messages kept is configurable. I've set it to keep logs for about 6 hours right now (a message is posted for each intent convergence every 30 seconds).

### CODE REVIEW

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt [hunk 1]
```diff
@@ -30,6 +33,24 @@ data class IntentActivity(
...
+ * @param actor Who (or what) initiated the operation.
+ */
+data class IntentConvergenceRecord(
+  val intentId: String,
+  val changeType: ChangeType,
+  val orchestrations: List<String>?,
+  val messages: List<String>?,
+  val diff: Set<FieldState>,
+  val actor: String,
+  val timeMs: Long
```
Comment (ajordens) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt]: If we're going full on javadoc ... do we want to include a definition of `timeMs`?

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt [hunk 2]
```diff
@@ -48,6 +69,18 @@ interface IntentActivityRepository {
...
+
+  fun getConvergenceLog(intentId: String): List<IntentConvergenceRecord>
+
+  /**
+   * Permalink to a specific log message, identified by timeMs
+   * @param intentId The ID of the intent that is being evaluated.
+   * @param timeMs The timestamp of the log message,
+   *  used as the unique identifier of the message.
+   */
+  fun getConvergenceLogEntry(intentId: String, timeMs: Long): List<IntentConvergenceRecord>
```
Comment (ajordens) [inline keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt]: Slightly confusing name ... get*Entry implies a single result but a `List` is returned?

File: keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/IntentController.kt [hunk 3]
```diff
@@ -116,6 +116,13 @@ class IntentController
 
   @RequestMapping(value = "/{id}/traces", method = arrayOf(RequestMethod.GET))
   fun getIntentTrace(@PathVariable("id") id: String) = traceRepository.getForIntent(id)
+
+  @RequestMapping(value = "/{id}/convergenceLog", method = arrayOf(RequestMethod.GET))
```
Comment (ajordens) [inline keel-web/src/main/kotlin/com/netflix/spinnaker/keel/controllers/v1/IntentController.kt]: Is `convergenceLog` being overly verbose ... could we just call it `/logs`?

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt (hunk 1 above)
Comment (emjburns) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt]: I'll change to timestampMillis also following our discussion on unit naming

File: keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt (hunk 2 above)
Comment (emjburns) [reply keel-core/src/main/kotlin/com/netflix/spinnaker/keel/IntentActivityRepository.kt]: There shouldn't be more than one here. Changing to return the first object if there is more than one with the same timestamp and then log a message . This shouldn't happen, and if it does it indicates that something is wrong with the queue processing.

**Pull request merged**